pkg/controller/cronjob/cronjob_controller_test.go

```diff
@@ -198,38 +198,38 @@ func TestSyncOne_RunOrNot(t *testing.T) {
-		"prev ran but done, is time, R":                 {R, F, onTheHour, noDead, T, F, justAfterTheHour(), T, F, 1, 0},
-		"prev ran but done, is time, suspended":         {A, T, onTheHour, noDead, T, F, justAfterTheHour(), F, F, 0, 0},
-		"prev ran but done, is time, past deadline":     {A, F, onTheHour, shortDead, T, F, justAfterTheHour(), F, F, 0, 0},
-		"prev ran but done, is time, not past deadline": {A, F, onTheHour, longDead, T, F, justAfterTheHour(), T, F, 1, 0},
-
-		"still active, not time, A":                {A, F, onTheHour, noDead, T, T, justBeforeTheHour(), F, F, 1, 0},
-		"still active, not time, F":                {f, F, onTheHour, noDead, T, T, justBeforeTheHour(), F, F, 1, 0},
-		"still active, not time, R":                {R, F, onTheHour, noDead, T, T, justBeforeTheHour(), F, F, 1, 0},
-		"still active, is time, A":                 {A, F, onTheHour, noDead, T, T, justAfterTheHour(), T, F, 2, 0},
-		"still active, is time, F":                 {f, F, onTheHour, noDead, T, T, justAfterTheHour(), F, F, 1, 0},
-		"still active, is time, R":                 {R, F, onTheHour, noDead, T, T, justAfterTheHour(), T, T, 1, 0},
-		"still active, is time, suspended":         {A, T, onTheHour, noDead, T, T, justAfterTheHour(), F, F, 1, 0},
-		"still active, is time, past deadline":     {A, F, onTheHour, shortDead, T, T, justAfterTheHour(), F, F, 1, 0},
-		"still active, is time, not past deadline": {A, F, onTheHour, longDead, T, T, justAfterTheHour(), T, F, 2, 0},
+		"never ran, not valid schedule, A":                           {A, F, errorSchedule, noDead, F, F, justBeforeTheHour(), F, F, 0, 1},
+		"never ran, not valid schedule, F":                           {f, F, errorSchedule, noDead, F, F, justBeforeTheHour(), F, F, 0, 1},
+		"never ran, not valid schedule, R":                           {f, F, errorSchedule, noDead, F, F, justBeforeTheHour(), F, F, 0, 1},
+		"never ran, not time, A":                                     {A, F, onTheHour, noDead, F, F, justBeforeTheHour(), F, F, 0, 0},
+		"never ran, not time, F":                                     {f, F, onTheHour, noDead, F, F, justBeforeTheHour(), F, F, 0, 0},
+		"never ran, not time, R":                                     {R, F, onTheHour, noDead, F, F, justBeforeTheHour(), F, F, 0, 0},
+		"never ran, is time, A":                                      {A, F, onTheHour, noDead, F, F, justAfterTheHour(), T, F, 1, 0},
+		"never ran, is time, F":                                      {f, F, onTheHour, noDead, F, F, justAfterTheHour(), T, F, 1, 0},
+		"never ran, is time, R":                                      {R, F, onTheHour, noDead, F, F, justAfterTheHour(), T, F, 1, 0},
+		"never ran, is time, suspended":                              {A, T, onTheHour, noDead, F, F, justAfterTheHour(), F, F, 0, 0},
+		"never ran, is time, past deadline but before another start": {A, F, onTheHour, shortDead, F, F, justAfterTheHour(), F, F, 0, 1},
```

Comment:
Naively, is the `but before another start` relevant?

Reply:
No, i'll get rid of that